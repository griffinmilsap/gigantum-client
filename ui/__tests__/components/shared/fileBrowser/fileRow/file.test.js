// vendor
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import relayTestingUtils from '@gigantum/relay-testing-utils';
import path from 'path';
// data
import codeData from 'Tests/components/labbook/code/__relaydata__/CodeBrowser.json';
// components
import File from 'Components/shared/fileBrowser/fileRow/File';


let edge = codeData.data.labbook.code.allFiles.edges[2]
let filename = path.basename(edge.node.key);
let file = {
  edge,
};

const mutationData = {
  owner: 'uitest',
  labbookName: 'ui-test-project',
  section: 'code',
  connection: 'CodeBrowser_allFiles',
  favoriteConnection: 'CodeFavorite_favorites',
  parentId: codeData.data.labbook.id,
};

let fixtures = {
  connectDragSource: jsx => jsx,
  closeLinkModal: jest.fn(),
  setState: jest.fn(),
  updateChildState: jest.fn(),
  codeDirUpload: jest.fn(),
  isSelected: true,
  filename,
  key: file.edge.node.key,
  multiSelect: false,
  mutationData,
  fileData: file,
  mutations: {
    moveLabbookFile: jest.fn(),
  },
  sort: 'all',
  reverse: true,
  childrenState: {},
  section: 'code',
  isDragging: false,
};

describe('Project File component', () => {

  it('File Snapshot', () => {
    const component = renderer.create(
       <File.DecoratedComponent {...fixtures}/>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  const component = mount(
     <File.DecoratedComponent {...fixtures}/>
  );

  it('Rename', () => {
    const evt = {
      target: {
        value: 'NewFileName.js',
      },
    };
    component.find('.ActionsMenu__item--rename').simulate('click');
    component.find('.File__input').simulate('change', evt);
    expect(component.state('newFileName')).toEqual(evt.target.value);
  });

  it('Cancel rename', () => {
    component.find('.File__btn--rename-cancel').simulate('click');
    expect(component.state('newFileName')).toEqual(fixtures.filename);
  });


  it('Rename add', () => {
    const evt = {
      target: {
        value: 'NewFileName.js',
      },
    };
    component.find('.ActionsMenu__item--rename').simulate('click');
    component.find('.File__input').simulate('change', evt);
    component.find('.File__btn--rename-add').simulate('click');
    expect(fixtures.mutations.moveLabbookFile.mock.calls.length).toEqual(1);
  });

  it('Mouseover file row', () => {
    let evt = {
      preventDefault: () => {},
    }
    component.find('.File').simulate('mouseover', evt);
    expect(component.state('hover')).toEqual(true);
  });


  it('Mouseout file row', () => {
    let evt = {
      preventDefault: () => {},
    }
    component.find('.File').simulate('mouseout', evt);
    expect(component.state('hover')).toEqual(false);
  });

  it('MouseEnter', () => {
    let evt = {
      preventDefault: () => {},
    }
    component.setProps({ isDragging: true });
    component.setState({ isDragging: true, isHovered: true });
    component.find('.File').simulate('mouseEnter', evt);
    expect(component.state('isDragging')).toEqual(true);
  });


  it('MouseLeave', () => {
    let evt = {
      preventDefault: () => {},
    }
    component.setProps({ isDragging: false });
    component.setState({ isDragging: false, isHovered: false });
    component.find('.File').simulate('mouseLeave', evt);
    expect(component.state('isDragging')).toEqual(false);
  });

  it('Set File to unchecked', () => {
    component.find('.Btn__check').simulate('click');
    expect(component.state('isSelected')).toEqual(false);
  });

  it('Set File to checked', () => {
    component.find('.Btn__uncheck').simulate('click');
    expect(component.state('isSelected')).toEqual(true);
  });
});
