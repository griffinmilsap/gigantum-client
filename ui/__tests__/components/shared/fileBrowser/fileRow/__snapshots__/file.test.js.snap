// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project File component File Snapshot 1`] = `
<div
  className="File"
  data-selenium-id="File"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="File__row"
    style={
      Object {
        "paddingLeft": "NaNpx",
      }
    }
  >
    <button
      className="Btn Btn--round Btn--medium Btn__check"
      onClick={[Function]}
    />
    <div
      className="File__cell File__cell--name"
    >
      <div
        className="File__icon book-icon"
      />
      <div
        className="File__text"
      />
    </div>
    <div
      className="File__cell File__cell--edit hidden"
    >
      <div
        className="File__container"
      >
        <input
          className="File__input"
          draggable={true}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onKeyDown={[Function]}
          type="text"
          value="README.md"
        />
      </div>
      <div
        className="flex justify-space-around"
      >
        <button
          className="File__btn--round File__btn--cancel File__btn--rename-cancel"
          onClick={[Function]}
        />
        <button
          className="File__btn--round File__btn--add File__btn--rename-add"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="File__cell File__cell--size"
    >
      3.3 kB
    </div>
    <div
      className="File__cell File__cell--date"
    >
      Fri Jan 18 2019
    </div>
    <div
      className="File__cell File__cell--menu"
    >
      <div
        className="ActionsMenu"
        data-selenium-id="ActionsMenu"
      >
        <button
          className="ActionsMenu__item Tooltip-data Tooltip-data--small Btn--round Btn--bordered Btn__addFolder visibility-hidden"
          data-tooltip="Add Subfolder"
          onClick={[Function]}
          type="button"
        />
        <div
          className="relative"
        >
          <button
            className="ActionsMenu__item Btn__delete Btn--noShadow Btn--round Btn--bordered Tooltip-data Tooltip-data--small"
            data-tooltip="Delete"
            disabled={false}
            onClick={[Function]}
            type="button"
          />
          <div
            className="ActionsMenu__popup hidden Tooltip__message"
          >
            <div
              className="Tooltip__pointer"
            />
            <p>
              Are you sure?
            </p>
            <div
              className="flex justify--space-around"
            >
              <button
                className="File__btn--round File__btn--cancel"
                onClick={[Function]}
                type="button"
              />
              <button
                className="File__btn--round File__btn--add"
                onClick={[Function]}
                type="button"
              />
            </div>
          </div>
        </div>
        <button
          className="ActionsMenu__item Btn__rename Btn--noShadow Tooltip-data Tooltip-data--small Btn--bordered Btn--round"
          data-tooltip="Rename"
          disabled={false}
          onClick={[Function]}
          type="button"
        />
        <button
          className="ActionsMenu__item Tooltip-data Tooltip-data--small Btn--round Btn--bordered Btn__Favorite-off Btn--noShadow"
          data-tooltip="Favorite"
          onClick={[Function]}
          type="button"
        />
      </div>
    </div>
  </div>
</div>
`;
